TombEngine/Objects/TR5/Entity/tr5_twogunlaser.cpp

```diff
@@ -0,0 +1,475 @@
+	{
+		auto* item = &g_Level.Items[itemNumber];
+
+		ClearItem(itemNumber);
+		SetAnimation(item, 6);
+	}
+
+	void FireTwogunWeapon(short itemNumber, short LeftRight, short plasma)
+	{
+		auto* item = &g_Level.Items[itemNumber];
+
+		auto pos1 = GetJointPosition(item, TWGunMesh[LeftRight], TWGunPositions[LeftRight]);
+		auto pos2 = GetJointPosition(item, TWGunMesh[LeftRight], Vector3i(TWGunPositions[LeftRight].x, TWGunPositions[LeftRight].y + 4096, TWGunPositions[LeftRight].z));
+		auto orient = Geometry::GetOrientToPoint(pos1.ToVector3(), pos2.ToVector3());
+
+		if (!plasma)
+		{
+			TriggerLaserBeam(pos1, pos2, LeftRight);
+
+			item->ItemFlags[LeftRight] = 16;
+			TriggerTwogunPlasma(pos1, Pose(pos1.ToVector3(), orient), 16);
+			TriggerTwogunPlasma(pos1, Pose(pos1.ToVector3(), orient), 16);
+			TriggerTwogunPlasma(pos1, Pose(pos1.ToVector3(), orient), 16);
+
+			Vector3i hitPos;
+			MESH_INFO* hitMesh = nullptr;
+
+			GameVector start = GameVector(pos1, 0);
+			GameVector end = GameVector(pos2, 0);
+			start.RoomNumber = item->RoomNumber;
+
+			if (ObjectOnLOS2(&start, &end, &hitPos, &hitMesh, ID_LARA) == GetLaraInfo(LaraItem)->ItemNumber)
+			{
+				if (LaraItem->HitPoints < 501)
+				{
+					ItemCustomBurn(LaraItem, Vector3(0.2f, 0.4f, 1.0f), Vector3(0.2f, 0.3f, 0.8f), 1 * FPS);
+					DoDamage(LaraItem, INT_MAX);
+				}
+				else
+					DoDamage(LaraItem, 250);
+			}
+		}
+	
+			return;
+
+			TriggerTwogunPlasma(pos1, Pose(pos1.ToVector3(), orient), abs(item->ItemFlags[LeftRight]));
+	}
+	
+
+	void TriggerTwogunPlasma(const Vector3i& posr, const Pose& pos, float life)
+	{
+		auto* sptr = GetFreeParticle();
+
+		sptr->on = true;
+
+		sptr->sB = (((GetRandomControl() & 127) + 128) * life) /16;
+		sptr->sR = sptr->sB - (sptr->sB >> 2);
+		sptr->sG = sptr->sR;
+		sptr->dR = 0;
+		sptr->dB = (((GetRandomControl() & 127) + 32) * life) /16;
+		sptr->dG = sptr->dB >> 2;
+		sptr->colFadeSpeed = 8 + (GetRandomControl() & 3);
+		sptr->fadeToBlack = 8;
+		sptr->sLife = sptr->life = (GetRandomControl() & 3) + 24;
+
+		sptr->blendMode = BLEND_MODES::BLENDMODE_ADDITIVE;
+		sptr->x = pos.Position.x;
+		sptr->y = pos.Position.y;
+		sptr->z = pos.Position.z;
+
+		int size = ((life * 64) * phd_cos(TO_RAD(pos.Orientation.x))) / 5;
+
+		sptr->xVel =  Random::GenerateInt(-128, 128) / 5;
+		sptr->yVel = life * 16 / 5;
+		sptr->zVel =  Random::GenerateInt(-128, 128) / 5;
+		sptr->friction = 0;
+		sptr->flags = SP_SCALE | SP_DEF | SP_ROTATE | SP_EXPDEF;
+		sptr->rotAng = GetRandomControl() & 4095;
+		sptr->rotAdd = (GetRandomControl() & 127) - 64;
+		sptr->gravity = (GetRandomControl() & 31) + 32;
+		sptr->maxYvel = 0;
+		sptr->scalar = 1;
+		sptr->size = sptr->sSize = size;
+		sptr->dSize = 1;
+	}
+
+	void TwogunControl(short itemNumber)
+	{
+		if (!CreatureActive(itemNumber))
+			return;
+
+		short angle, head, xTorso, yTorso, frame, base, roomNumber;
+
+		auto* item = &g_Level.Items[itemNumber];
+		auto* creature = GetCreatureInfo(item);
+
+		angle = 0;
```

Comment:
Can you move these 4 init lines upwards, next to a declaration of these variables? This will make things cleaner.